plenum/common/messages/fields.py

```diff
@@ -376,6 +376,27 @@ def _specific_validation(self, val):
             return 'empty serialized value'
 
 
+class VersionField(FieldBase):
+    _base_types = (str,)
+
+    def __init__(self, components_number=(3,), **kwargs):
+        super().__init__(**kwargs)
+        self._comp_num = components_number
+
+    def _specific_validation(self, val):
+        parts = val.split(".")
+        if len(parts) not in self._comp_num:
+            return "version consists of {} components, but it should contain {}".format(len(parts), self._comp_num)
+        for p in parts:
+            try:
+                convertedint = int(p)
```

Comment:
Not sure but "            0.1        .2" (white spaces before/after) maybe pass too.